### PR TITLE
remove underscore dependency

### PR DESCRIPTION
**Pull request by 1067**
remove/inline underscore deps

### CODE REVIEW

File: callbacks.js [hunk 1]
```diff
@@ -93,7 +94,7 @@
          * @returns {Callbacks}
          */
         remove: function(handler, scope) {
-            if (_.isFunction(handler)) {
+            if (typeof handler == 'function' || false) {
```
Comment (Metallium) [inline callbacks.js]: false is redundant?

File: callbacks.js [hunk 2]
```diff
@@ -135,9 +145,13 @@
         },
 
         _removeScope: function(scope) {
-            this._subscriptions = _.reject(this._subscriptions, function(subscription) {
-                return subscription.scope === scope;
-            });
+            this._subscriptions = this._subscriptions
+                .reduce(function(memo, subscription) {
```
Comment (Metallium) [inline callbacks.js]: .filter by  !==

File: callbacks.js [hunk 3]
```diff
@@ -121,10 +122,19 @@
...
         _findHandlerIndex: function(handler, scope) {
-            return _.findIndex(this._subscriptions, function(subscription) {
+            var predicate = function (subscription) {
                 return subscription.scope === scope &&
                     (subscription.handler === handler || subscription.originalHandler === handler);
-            });
+            };
+
+            var length = this._subscriptions.length;
+            for (var index = 0; index >= 0 && index < length; index += 1) {
```
Comment (Metallium) [inline callbacks.js]: extract to separate method

File: callbacks.js [hunk 4]
```diff
@@ -135,9 +136,21 @@
...
-            });
+            this._subscriptions = this._subscriptions
+                .filter(function(subscription) {
+                    return subscription.scope !== scope;
+                });
+        },
+
+        _findIndex: function(array, predicate) {
+            var length = array.length;
+            for (var index = 0; index < length; index += 1) {
```
Comment (victor-homyakov) [inline callbacks.js]: Merge two lines into one:

    for (var index = 0, length = array.length; index < length; index += 1) {

**Pull request merged**